nomenclature/core.py

```diff
@@ -52,21 +51,22 @@ def process(
     ValueError
         If the :class:`pyam.IamDataFrame` fails the validation.
     """
-    # The deep copy is needed so we don't alter dsd in dimensions.remove("region")
-    dimensions = copy.deepcopy(dimensions or dsd.dimensions)
-
-    # perform validation and region-processing (optional)
-    if processor is None:
-        dsd.validate(df, dimensions=dimensions)
-    else:
-        if "region" in dimensions:
-            dimensions.remove("region")
-            dsd.validate(df, dimensions=dimensions)
-            df = processor.apply(df)
-            dsd.validate(df, dimensions=["region"])
-        else:
-            dsd.validate(df, dimensions=dimensions)
-            df = processor.apply(df)
+
+    processor = processor or []
+    processor = processor if isinstance(processor, list) else [processor]
+
+    dimensions = dimensions or dsd.dimensions
+
+    if (
+        any(isinstance(p, RegionProcessor) for p in processor)
+        and "region" in dimensions
+    ):
+        dimensions.remove("region")
```

Comment:
> This would remove region even if explicitly stated
True, but is that a bad thing since `RegionProcessor` validates the regions anyway?

Reply:
Ok, fair enough...

Reply:
But please add a comment why the "region" dimension is removed.

Reply:
Looking at it again, it might be a good idea to remove the `dimensions` argument from the `process` function altogether. We can influence which dimension in being validated at creation of the `DataStructureDefinition`.
Is there a use case where we want to have a `DataStructureDefinition` with say _region_, _variable_ and _scenario_ but we only want to validate _region_ and _variable_?

Reply:
> But please add a comment why the "region" dimension is removed.

Sure